src/lib/propublica.ts

```diff
@@ -1 +1,141 @@
-// propublica api
\ No newline at end of file
+// propublica api
+
+import axios from 'axios';
+
+type Vote = {
+    chamber: string;
+    date: string;
+    time: string;
+    roll_call: string;
+    question: string;
+    results: string;
+    total_yes: number;
+    total_no: number;
+    total_not_voting: number;
+    api_url: string;
+};
+
+export function getSubjectUrl(topic: string): Promise<string> {
+    const apiUrl = `https://api.propublica.org/congress/v1/bills/subjects/search.json?query=${topic}`;
+    return axios
+        .get(apiUrl, {
+            headers: {
+                'X-API-Key': process.env.PROPUBLICA_API_KEY,
+            },
+        })
+        .then((response) => {
+            // eslint-disable-next-line @typescript-eslint/no-unsafe-member-access
+            return response.data.results[0].subjects[0].url_name as string;
+        })
+        .catch((error) => {
+            return Promise.reject(error);
+        });
+}
```

Comment:
There's a few optimizations we can make here. 
```ts
// SubjectResponse is the type of the response
return axios.get<SubjectResponse>(apiUrl, { headers: { 'X-API-Key': env.PROPUBLICA_API_KEY }});
```

This way we know the type of the response and don't have to use any eslint disable's.  

We for the most part try to only use async/await syntax as opposed to chaining .then's.  For consistency's sake, please convert the code to use async/await.

Also, we use the env slightly different on the backend.  You need to add the appropriate types [here](https://github.com/c2c-project/prytaneum-townhall/blob/master/src/config/env/env.ts), then `import env from 'config/env'` and access environment variables like that as opposed to directly from `process.env`.  The reasoning is so that way we have run time defaults of environment variables.

Please also make the appropriate changes to other parts of this file with respect to this comment.

Reply:
Regarding the env, should i commit the config/env.ts file even if I add the api key there?